cocos/renderer/gfx-gles2/GLES2Context.h

```diff
@@ -107,6 +107,7 @@ class CC_GLES2_API GLES2Context final : public Context {
     EGLSurface _eglSurface = EGL_NO_SURFACE;
     EGLContext _eglContext = EGL_NO_CONTEXT;
     EGLContext _eglSharedContext = EGL_NO_CONTEXT;
+    EGLConfig *_eglConfigList = nullptr;
```

Comment:
`std::vector<EGLConfig>` would be better

Reply:
just curious why vector is better here, the _eglConfigList size is fixed once created, and seems vector will introduce new std header files in

Reply:
https://clang.llvm.org/extra/clang-tidy/checks/modernize-avoid-c-arrays.html

Reply:
tried vector,
`eglChooseConfig(_eglDisplay, defaultAttribs, _eglConfigList, count, &numConfig)` requires an input-output memory pointer, vector.data() will replace _eglConfigList here but this is not how vector works, cuz we update data with raw pointer hold by this vector without updating its size or capacity, etc. we can do sth hack here make it work, just wondering whether worth it.

Reply:
https://www.cplusplus.com/reference/vector/vector/resize/

Reply:
i was using reserve(size), not saying its feasible or not, i mean this usage is not vector designed for.

Reply:
No, it does. Actually, it's quite common to use it this way. And `resize` is the more proper choice than `reserve`, since `reserve` doesn't actually bump the vector size, and it'll introduce problems when using the `.size()` function afterward.